src/App.js

```diff
@@ -4,18 +4,30 @@ import AddTodo from './components/AddTodo'
 import Todo from './components/Todo'
 import Footer from './components/Footer'
 import './App.css'
+
+
 export default function App() {
     const [todoItems, setTodoItems] = useState([])
+    const [editId, setEditId] = useState()
+
+    function editTodo(id) {
+        setEditId(id)
+    }
 
     return (
         <div className="App">
             <Header />
             <main>
                 <AddTodo
                     setTodoItems={setTodoItems}
+                    editId={editId}
+                    setEditId={setEditId}
+                    itemToEdit={editId ? todoItems.find((e) => e.id === editId) :undefined}
```

Comment:
You now do not need to pass in `editId` because `itemToEdit` lets `AddTodo` know that we are trying to edit a todo list item.

Reply:
So, inadvertently, instead of saving just the id of the element you want to edit in state, save the whole element.

`const [itemToEdit, setItemToEdit] = useState(null)` instead of what you have on line 11